src/transports/smart_protocol.c

```diff
@@ -298,7 +301,21 @@ static int fetch_setup_walk(git_revwalk **out, git_repository *repo)
 		if (git_reference_type(ref) == GIT_REF_SYMBOLIC)
 			continue;
 
-		if ((error = git_revwalk_push(walk, git_reference_target(ref))) < 0)
+                /* only walk over commits */
```

Comment:
The comment here is a bit terse, in my opinion. I'd love to have it mention why we need the separate lookup/peel steps and why we need to filter out tags-to-blobs here if there's a comment 10 lines before which pretends to filter out tags already. I've been confused by that, as well, as I didn't expect you to have tags outside of the "refs/tags" hierarchy.